src/main.rs

```diff
@@ -1,3 +1,4 @@
+#![warn(clippy::all)]
```

Comment:
I think this attribute is redundant. As per the [docs](https://github.com/rust-lang/rust-clippy/blob/b4e4fa51ea328adf5b182d3820dca9286d5ef3bf/README.md#clippy) state:
> * `clippy::all` (everything that is on by default: all the categories below except for `nursery`, `pedantic`, and `cargo`)

```suggestion
```

Reply:
You are right that it is unnecessary for running clippy, but adding it explicit turns on clippy notes for IDEs like VS Code (if you use the default of opt-in), so I'd prefer to keep it.

Reply:
Fair enough 👍 That simply works.

You know.. it seldom feels right to have these "breadcrumbs" of tooling scattered around the code. I did find a couple ways to configure vscode but alas, they're kinda clumsy.

Reply:
Yeah, not a big fan myself. I'm more ok with clippy I guess as you could consider it rust infrastructure, and explicit being better than implicit and all that but I'd rather it be something we could keep on that multi-IDE configuration file you mentioned hehe